survey_formio/models/survey.py

```diff
@@ -0,0 +1,152 @@
+        return "form"
+
+    def _get_formio_components(self):
+        """
+        Pages are not supported yet.
+
+        Returns a list of dict representing survey questions as form.io components
+        """
+        self.ensure_one()
+        components = []
+        for question in self.question_ids:
+            component = question._get_formio_component()
+            if component:
+                components.append(component)
+        return components
+
+    def _from_formio_datetime_to_odoo(self, datetime_str):
+        """
+        Form.io datetimes are given in the user's timezone and with an offset.
+        Odoo wants naïve datetimes.
+
+        Returns a naïve datetime as a string
+
+        Parameters:
+        datetime_str: string representing a datetime with an offset
+        """
+        # In the json given by form.io, the offset is formatted as +hh:mm
+        # To be able to use strptime we need it to be formated as +hhmm
+
+        # We can't use the fromisoformat function because it's only available
+        # in Python v3.7+ and Odoo v13 is compatible with Python v3.6
+        index = datetime_str.rfind(":")
+        datetime_str = datetime_str[:index] + datetime_str[index + 1 :]
+        utc_datetime = datetime.strptime(
+            datetime_str, "%Y-%m-%dT%H:%M:%S%z"
+        ).astimezone(pytz.UTC)
+        return utc_datetime.strftime("%Y-%m-%d %H:%M:%S")
+
+    def _from_formio_multiple_choice_to_odoo(
+        self, question_id, multiple_choice_answers_dict
+    ):
+        """
+        For a multiple choice question, the form.io answer has the
+        following format:
+        {"question_id": {
+          "answer_1_value": true/false
+          "answer_2_value": true/false
+          ....
+        }}
+        Odoo expects the following format:
+        {"question_id_answer_id": "answer_id"}
+
+        Returns a dictionnary with the correct format.
+
+        Parameters:
+        question_id: id of the survey.question
+        multiple_choice_answers_dict: dict given by form.io for a multiple choice question
+        """
+        answers_dict = {}
+        for answer in multiple_choice_answers_dict.keys():
+            answers_dict[question_id + "_" + answer[1:]] = answer[1:]
+        return answers_dict
+
+    def generate_formio_json(self):
+        """
+        Returns a form.io conform json representation of a survey.
+        """
+        self.ensure_one()
+        display = self._get_formio_display()
+        title = self.title
+        components = self._get_formio_components()
+
+        form_json = json.dumps(
+            {"display": display, "title": title, "components": components}
+        )
+        return form_json
+
+    def user_input_from_formio(self, formio_input, user_input_id=None):
+        """
+        If no user_input_id is given, creates a user_input with the values in formio_input.
+        If there is a user_input_id, updates that user_input with the values in formio_input.
```

Comment:
FYI, OCA guidelines ask to use present imperative in comments: https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#directories:~:text=Use%20present%20imperative%20(Fix%20formatting%2C%20Remove%20unused%20field)%20avoid%20appending%20%27s%27%20to%20verbs%3A%20Fixes%2C%20Removes